### ISSUE TITLE
Execute CFAM to boot witherspoon

### ISSUE THREAD

**Issue by geissonator** [openbmc/openbmc#650]
A one liner we need in openbmc code to power on witherspoon.  It requires openFSI and basic scom/mbox support.

This is what we need to start the P9 boot process

putcfam pu 2801 80000000

**Comment by geissonator** [comment]
Looks like initially we're going to use pdbg to do power on work.  This will track us moving over to openFSI once ready.

**Comment by shenki** [comment]
@geissonator do we have a ticket to track that? 

Alistair is about to release a major update to pdbg that improves it's stability. We should target that version for power on.

**Comment by apopple** [comment]
Hey, for information once OpenFSI is ready I plan on making an OpenFSI backend for pdbg as well. This means the same commands should continue to work with pdbg but using the OpenFSI kernel driver (which will hopefully be quicker/more reliable than bit-banging from userspace :)

**Comment by geissonator** [comment]
Good stuff guys, this work is tracked in https://github.com/openbmc/openbmc/issues/647.  pdbg is currently only in the debug tarball, we're going to bring it into the main image with that issue and run the appropriate scripts to get power on working.

**Comment by GHF** [comment]
@geissonator @shenki How are you folks overriding the GPIOs used by pdbg? It looks like that's all hardcoded: https://github.com/open-power/pdbg/blob/master/libpdbg/bmcfsi.c#L31-L39

I don't see machine patches against pdbg in meta-witherspoon.

**Comment by shenki** [comment]
We chatted on IRC about this. The goal is to get openfsi merged into the tree, and have pdbg use that.

If we don't have that in place before the end of the week we will instead apply a patch to pdbg so it can be used in the mean time.

**Comment by apopple** [comment]
@shenki You will also need a direct FSI read/write interface for OpenFSI to kick off the CFAM sequence. Chris was going add this to OpenFSI at some point, I will check on the current patch series. It should be pretty easy to add if it's not there though.

**Comment by geissonator** [comment]
This was done using pdbg (thanks @apopple!).  #866 tracks us moving over directly to openFSI where applicable.
